helmfile/charts/gatekeeper/templates/policies/disallow-localhost-seccomp.rego

```diff
@@ -0,0 +1,181 @@
+package k8sdisallowlocalhostseccomp
+
+# For pods
+
+# Violation if pods use Localhost seccompProfile
+violation[{"msg": msg}] {
+
+    object := get_pods
+
+    object.spec.securityContext.seccompProfile.type == "Localhost"
+
+    msg := sprintf("The %v named <%v> uses Localhost seccompProfile. Elastisys Compliant Kubernetes does not allow the use of Localhost secompProfile.", [object.kind, object.metadata.name])
+}
+
+# Get "Pods"
+get_pods = res {
+    input.review.object.kind == "Pod"
+    res := input.review.object
+}
+
+# For deployments, statefulsets, daemonsets, replicasets, jobs, replicationcontrollers
+
+# Violation if resources that uses podTemplates use Localhost seccompProfile
+violation[{"msg": msg}] {
+
+    object := get_objects
+
+    object.spec.template.spec.securityContext.seccompProfile.type == "Localhost"
+
+    msg := sprintf("The %v named <%v> uses Localhost seccompProfile in the pod template. Elastisys Compliant Kubernetes does not allow the use of Localhost secompProfile.", [object.kind, object.metadata.name])
+}
+
+# Get resources that use pod templates.
+get_objects = res {
+    kinds := [
+        "Deployment",
+        "StatefulSet",
+        "DaemonSet",
+        "ReplicaSet",
+        "Job",
+        "ReplicationController"
+    ]
+    input.review.object.kind == kinds[_]
+
+    res := input.review.object
+}
+
+# For cronjobs
+
+# Violation cronjobs use Localhost seccompProfile
+violation[{"msg": msg}] {
+
+    object := get_cronjobs
+
+    object.spec.jobTemplate.spec.template.spec.securityContext.seccompProfile.type == "Localhost"
+
+    msg := sprintf("The %v named <%v> uses Localhost seccompProfile in the pod template. Elastisys Compliant Kubernetes does not allow the use of Localhost secompProfile.", [object.kind, object.metadata.name])
+}
+
+# Get "CronJobs"
+get_cronjobs = res {
+    input.review.object.kind == "CronJob"
+    res := input.review.object
+}
+
+# For regular containers
+
+# Violation if containers use Localhost seccompProfile
+violation[{"msg": msg}] {
+
+    container := get_containers[_]
+
+    container.securityContext.seccompProfile.type == "Localhost"
+
+    msg := sprintf("The container named <%v> uses Localhost seccompProfile. Elastisys Compliant Kubernetes does not allow the use of Localhost secompProfile.", [container.name])
+}
+
+# Get containers for "Pods"
+get_containers = res {
+    input.review.object.kind == "Pod"
+    res := input.review.object.spec.containers
+}
+
+# Get containers for resources that use pod templates.
+get_containers = res {
+    kinds := [
+        "Deployment",
+        "StatefulSet",
+        "DaemonSet",
+        "ReplicaSet",
+        "Job",
+        "ReplicationController"
+    ]
+    input.review.object.kind == kinds[_]
+
+    res := input.review.object.spec.template.spec.containers
+}
```

Comment:
Have you tried using the `get_objects` function and the extracting containers that way? So that retrieving `pod templates` are done in one place :slightly_smiling_face:

Reply:
:+1: Nice improvement, I made another commit reusing some of the functions i defined above.